src/profile-logic/profile-data.js

```diff
@@ -150,6 +150,186 @@ export function getSampleCallNodes(
   });
 }
 
+// This function assumes that the path isn't empty
+export function getFuncIndex(path: CallNodePath): IndexIntoFuncTable {
+  return path[path.length - 1];
+}
+
+// Returns the JS implementation information for a specific stack
+export function getJsImplementationForStack(
+  stackIndex: IndexIntoStackTable,
+  { stackTable, frameTable, stringTable }: Thread
+): string {
+  const frameIndex = stackTable.frame[stackIndex];
+  const jsImplementationStrIndex = frameTable.implementation[frameIndex];
+  const jsImplementation =
+    jsImplementationStrIndex === null
+      ? 'interpreter'
+      : stringTable.getString(jsImplementationStrIndex);
+
+  return jsImplementation;
+}
+
+type Breakdown = { [key: string]: Milliseconds } | null;
+type ItemTimings = {|
+  selfTime: {|
+    // time spent excluding children
+    value: Milliseconds,
+    breakdownByImplementation: Breakdown,
+  |},
+  totalTime: {|
+    // time spent including children
+    value: Milliseconds,
+    breakdownByImplementation: Breakdown,
+  |},
+|};
+
+export type TimingsForPath = {|
+  // timings for this path
+  forPath: ItemTimings,
+  // timings for this func across the tree
+  forFunc: ItemTimings,
+  rootTime: Milliseconds, // time for all the samples in the current tree
+|};
+
+// Returns the timings for a specific path. The algorithm is adjusted when
+// the call tree is inverted.
+export function getTimingsForPath(
+  path: CallNodePath,
+  { callNodeTable, stackIndexToCallNodeIndex }: CallNodeInfo,
+  interval: number,
+  isInvertedTree: boolean,
+  thread: Thread
+): TimingsForPath {
+  const { samples, stackTable, funcTable } = thread;
+  const nodeIndex = getCallNodeIndexFromPath(path, callNodeTable);
+  const funcIndex = getFuncIndex(path);
```

Comment:
Yeah I'm not very convinced either. But I think it's still more meaningful than `path[path.length - 1]` when reading the code. The main question is: when looking at this function, can you infer what it means from its name of do you need to go and look at its (very simple) implementation?

I see the `path` to designate a specific node, that's why, for me, when reading `getFuncIndex(path)` it actually implies that I want the leaf function index. But if it's not obvious for you it means it's not obvious enough.

Please tell me what you think, I can go one way or another :)

Reply:
After a night, I started to understand why you want to add the "Leaf" word in many of the names: because you see a path as a sequence of func indices, while I see it as an address. I mean that I see it as an object with a meaning by itself while you don't think about it more than as this sequence.
All this to say that I now understand your point of view better and I'll add "Leaf" for this function.

Reply:
Hah, funny how our mental models can differ like that. Thanks.